### ISSUE TITLE
Re-added an expired Business plan but the plan shown in the menu shows as "Free"

### ISSUE THREAD

**Issue by mrfoxtalbot** [Automattic/wp-calypso#57325]
### Quick summary

The plan displayed next to "Upgrades" in the unified navigation is still showing the previous plan when navigation back from the upgrade process. It should update automatically to show the new plan.

### Steps to reproduce

1. Upgrade a site (I tried to upgrade from free to Business)
2. Click on "see my new plan features" to go back to the site's dashboard once the upgrade has been completed successfully.
3. Check the plan on the navigation bar, it is still showing "free".

![Screen Shot on 2021-10-25 at 18:52:22](https://user-images.githubusercontent.com/4452464/138738044-edc7ba26-7f33-40f3-b443-6c7331f037af.png)

This happened with two sites:

- mrfoxat02.wpcomstaging.com
- mrfoxat03.wpcomstaging.com

Here is the complete flow:

https://user-images.githubusercontent.com/4452464/138738852-6d552c6b-3d4c-48cf-afd5-abd33c11944b.mov

### What you expected to happen

It should be refreshed automatically to list the new plan.

### What actually happened

It did not.

### Context

Dogfooding the automatic downgrade and re-upgrade process.

### Operating System

macOS

### Browser

Google Chrome/Chromium

### Simple, Atomic or both?

Simple

### Theme-specific issue?

Nope

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

Yes, easy to implement

### Workaround details

Reloading the dashboard page in the browser will show the correct plan, but na

**Comment by DavidRothstein** [comment]
>### Simple, Atomic or both?
>
>Simple

I couldn't reproduce it with a simple site, and it looks from the sites you're testing that it's actually Atomic?  Is it the same issue as https://github.com/Automattic/wp-calypso/issues/53294?

**Comment by mrfoxtalbot** [comment]
Yep that seems to be a duplicate. I'm closing this one, thank you!

**Comment by JoshuaGoode** [comment]
For visibility, noting what I shared on the ongoing issue we kept open:

This is generally due to the `jetpack_active_plan` option not being properly set and should sync/fix itself but it's best to fix it for a user ASAP.

Internal notes: P9HQHe-14R-p2

**Workaround:** A quick way to get the proper `jetpack_active_plan` option is to navigate to My Site > Jetpack > Dashboard.  That will get the option updated.
